docs/developer-guide.md

```diff
@@ -1,8 +1,115 @@
 # Developer Guidelines
 
-## Setting up the commit-msg hook for the pre-commit
+## Setting up the Local Development Environment
```

Comment:
Should we separate the
- Developer Guidelines
- Development environment setup
into 2 separate .md files that we can refer to independently?